app/views/posts/index.html.erb

```diff
@@ -0,0 +1,28 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Members Only Posts</h1>
+
+<% @posts.each do |post| %>
+  <% if signed_in? %>
+    <p class="float-right">
+      Posted by:
+      <%= post.user.email %>
+    </p>
+    <h4 class="float-left"><%=post.title %></h4>
+    <p class="clear"><%= post.body %></p>
+    <p>
+    <%= link_to 'Edit', edit_post_path(post) %>
+    <%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %>
+    </p>
+  <% end %>
+
+<% end %>
+
+
+<% if signed_in? %>
+  <%= link_to "Create a Post", new_post_path %>
+<% end %>
```

Comment:
Kindly move this logic into the helper module. As much as possible, ensure that there's no logic in views.